src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -74,21 +104,48 @@ class MoneyRequestAmountPage extends React.Component {
     }
 
     componentDidMount() {
+        if (this.isEditing && (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0)) {
+            Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID));
```

Comment:
Shouldn't we return early here?

Reply:
Hmm, yeah I think it's better to return early.

Reply:
Also, reverted back the if condition.